src/me/eglp/gv2/util/base/guild/reminder/GuildReminder.java

```diff
@@ -0,0 +1,196 @@
+		this.channelID = channel.getID();
+		this.id = id;
+	}
+
+	public GuildReminder(GraphiteGuild guild, String id, String channelID, String message,
+			ReminderRepetitionEnum repeatMs, LocalDateTime date, LocalDateTime latestPossibleDate) {
+		this.guild = guild;
+		this.id = id;
+		this.channelID = channelID;
+		this.message = message;
+		this.repeatMs = repeatMs;
+		this.date = date;
+		this.latestPossibleDate = latestPossibleDate;
+	}
+
+	public String getMessage() {
+		return message;
+	}
+
+	public GraphiteGuild getGuild() {
+		return guild;
+	}
+
+	public String getChannelID() {
+		return channelID;
+	}
+
+	public String getId() {
+		return id;
+	}
+
+	public ReminderRepetitionEnum getRepeatMs() {
+		return repeatMs;
+	}
+
+	public LocalDateTime getDate() {
+		return date;
+	}
+
+	public LocalDateTime getLatestPossibleDate() {
+		return latestPossibleDate;
+	}
+
+	public ScheduledFuture<?> getFinishFuture() {
+		return finishFuture;
+	}
+
+	public void remove() {
+		try {
+			guild.getRemindersConfig().removeReminder(id);
+			if (finishFuture != null)
+				finishFuture.cancel(false);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	private void sendMessage() {
+		Graphite.withBot(Graphite.getGraphiteBot(), () -> {
+			GraphiteGuildMessageChannel messageChannel = guild.getGuildMessageChannelByID(channelID);
+			if (messageChannel == null) {
+				this.remove();
+				return;
+			}
+			try {
+				if (repeatMs != null) {
+					messageChannel.sendMessageComplete(repeatMs.getFriendlyName() + " Reminder: " + message);
+				} else {
+					messageChannel.sendMessageComplete("Simple Reminder: " + message);
+				}
+			} catch (Exception e) {
+				throw e;
+			}
+		});
+	}
+
+	private void calculateNextPossibleReminderDate() {
+		LocalDateTime now = LocalDateTime.now(guild.getConfig().getTimezone());// Current time with correct UTC for
+																				// guild
+		while (!latestPossibleDate.isAfter(now)) {
+			// maybe
+			latestPossibleDate = latestPossibleDate.plusYears(repeatMs.getYearsDisplacement());
+			latestPossibleDate = latestPossibleDate.plusMonths(repeatMs.getMonthsDisplacement());
+			latestPossibleDate = latestPossibleDate.plusWeeks(repeatMs.getWeeksDisplacement());
+			latestPossibleDate = latestPossibleDate.plusDays(repeatMs.getDaysDisplacement());
+		}
+	}
+
+	public void enqueue() {
+		finishFuture = Graphite.getScheduler().getExecutorService().schedule(() -> {
+			sendMessage();
+
+			if (repeatMs != null) {
+				// Todo reenqueue
+				calculateNextPossibleReminderDate();
+				enqueue();
+			} else {
+				// Todo remove from db
+				this.remove();
+			}
+		}, latestPossibleDate.atZone(guild.getConfig().getTimezone()).toEpochSecond() - Instant.now().getEpochSecond(),
+				TimeUnit.SECONDS);
+	}
+
+	public boolean load() {
+		if (this.repeatMs == null)
+			return false;
+		calculateNextPossibleReminderDate();
+		enqueue();
+		return true;
+	}
+
+	@Override
+	public void preSerializeWI(JSONObject object) {
```

Comment:
Lots of the types used here aren't actually serializable. Just leave the `WebinterfaceObject` implementation out for now, it can be added later when the functionality is added to the webinterface